docs/DesignDocs/AttestationApiRequirements.md

```diff
@@ -0,0 +1,291 @@
+Verifier, or Relying Party.  An Attester generates signed Evidence
+to be processed by a Verifier.  A Verifier appraises Evidence using
+an Appraisal Policy (and often Endorsements as well), and generates
+Attestation Results to be processed by a Relying Party.  A Relying
+Party appraises Attestation Results and typically uses them for
+some authorization decision.
+
+The APIs are thus centered around that conceptual data flow, as
+depicted in Figure 1 in the
+[RATS Architecture](https://tools.ietf.org/wg/rats/draft-ietf-rats-architecture/).
+
+Specific formats of Evidence, Attestation Results, etc. are the
+responsibility of the relevant plugins to implement, not the application.
+Similarly, plugins are responsible for TEE-specific details.
+
+The (non-attestation specific) protocol between the Attester and
+a Relying Party is the responsibility of the application, not
+the OE SDK or plugins.
+
+The (attestation-specific) protocol between a Verifier and either
+an Attester (in the Passport model) or a Relying Party (in the
+Background-check model) is the responsibility of the relevant
+plugins to implement, not the application.
+
+Specification
+-------------
+
+The APIs below are described as _abstract_ APIs, and any concrete
+APIs in C (or any other language) needs to accommodate an equivalent
+set of operations and support for an (at least) equivalent set of inputs
+and outputs.  This means the actual mapping of functions or arguments
+need not be 1:1 as long as a mapping exists.
+
+These APIs apply to both the "northbound" API exposed to apps by the SDK,
+as well as the "southbound" API between the SDK and plugins.
+
+# Attester APIs
+
+## GetEvidence call
+
+Inputs:
+
+* Requested Evidence format, if any (e.g., CWT, JWT, X.509, etc.).
+  Defaults to letting the plugin choose one.
+* Challenge buffer OCTET STRING
+* Include endorsements BOOLEAN
+* Custom claims buffer OCTET STRING
+
+Outputs:
+
+* Evidence buffer OCTET STRING
+* Format used for Evidence buffer
+
+Return status codes:
+
+* Success indicates that Evidence was generated as requested
+* Failed-to-get-endorsements indicates that the call failed because
+  "Include endorsements" was true but that endorsements could not be
+  obtained
+* Requested-format-not-supported indicates that the requested Evidence
+  format could not be used
+* Challenge-Parse-error indicates that the challenge buffer provided could not
+  be correctly parsed
+* Custom-Claims-Parse-error indicates that the custom claims buffer provided
+  could not be correctly parsed per the requested Evidence format
+* Other failure indicates that the call failed for any other reason
+
+This is the main attestation functionality of the Attester.
+The challenge buffer, which may be empty for some protocols (e.g.,
+those that use timestamps and synchronized clocks to check freshness),
+can be provided by the app if the app has its own way to get it
+from a Verifier.  For example, in the example in Section 15.4 of
```

Comment:
In the latest public version of [RATS Arch](https://tools.ietf.org/html/draft-ietf-rats-architecture-02) document, I did not find Section 15.4, nor definition of challenge. Are these in a newer internal draft?

Reply:
Ah.  See https://github.com/ietf-rats-wg/architecture which has links to Editors Copy and diffs from last I-D.

Reply:
Updated link here to point to the Editors Copy for now.